File: Leetcode/Easy/PYTHON/1523. Count Odd Numbers in an Interval Range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        return (high + 1) // 2 - low // 2

#Short Thought Process:
# To count odd numbers in the interval [low, high], we can use the formula: (high + 1) // 2 - low // 2.
# This formula works because (high + 1) // 2 gives the count of odd numbers from 1 to high,
# and low // 2 gives the count of odd numbers from 1 to low - 1.
# By subtracting these two counts, we get the count of odd numbers in the interval [low, high].\
    
# https://leetcode.com/problems/count-odd-numbers-in-an-interval-range/

File: Leetcode/Easy/CPP/1523. Count Odd Numbers in an Interval Range.cpp
class Solution {
public:
    int countOdds(int low, int high) {
        int count=0;
        for(int i=low;i<=high;i++){
            if(i%2!=0) count++;
        }
        return count;
    }
};

// Short Thought Process: To count the odd numbers in a given interval [low, high], we iterate through each number in the range and check if it is odd by using the modulus operator. If a number is odd, we increment our count. Finally, we return the total count of odd numbers found in the interval.

//Link: https://leetcode.com/problems/count-odd
// -numbers-in-an-interval-range/